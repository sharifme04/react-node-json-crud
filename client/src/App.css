.app-title{
  margin-top: 25px;
}

.search-field{
  width: 35%;
}