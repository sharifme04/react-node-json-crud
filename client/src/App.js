import React, { useEffect, useState } from "react";
//import axios from "axios";
import { Routes, Route } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import "./App.css";
import { getToDo } from "./app/actions/todo";
import Home from "./app/components/Home";
import Add from "./app/components/Add";
import Edit from "./app/components/Edit";

function App() {
  const dispatch = useDispatch();
  const { todo, isLoading } = useSelector((state) => state.todoReducer || {});
  const [value, setValue] = useState("");

  useEffect(() => {
    dispatch(getToDo());
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  const handleChange = (event) => {
    setValue(event.target.value);
  };

  return (
    <div className="container">
      <h2 className="app-title">Todo App</h2>
      <hr></hr>
      <Routes>
        <Route
          exact
          path="/"
          element={
            <Home
              todoList={todo}
              isLoading={isLoading}
              handleChange={handleChange}
              value={value}
            />
          }
        />
        <Route path="/todo/edit/:id" element={<Edit />} />
        <Route path="/todo/addnew" element={<Add />} />
      </Routes>
    </div>
  );
}

export default App;
