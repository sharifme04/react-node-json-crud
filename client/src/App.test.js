// import { render,fireEvent, screen } from '@testing-library/react';
//  import userEvent from '@testing-library/user-event'
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { MemoryRouter,Route, Routes} from 'react-router-dom'
import renderer from "react-test-renderer";
import App from './App';

const mockStore = configureMockStore();
const store = mockStore({})

test("snapshot data renders", () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter initialEntries={["/"]}>
        <App  />
      </MemoryRouter>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot renders when navigate add route", () => {
  let testHistory, testLocation;   // eslint-disable-line no-unused-vars
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter initialEntries={["/todo/addnew"]}>
        <App  />
        <Routes>
        <Route 
             path="*"
             render={({ history, location }) => {
              testHistory = history;
              testLocation = location;
              return null;
            }} />
            </Routes>
      </MemoryRouter>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot renders when navigate edit route", () => {
  let testHistory, testLocation;   // eslint-disable-line no-unused-vars
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter initialEntries={["/todo/edit/478951"]}>
        <App  />
        <Routes>
        <Route 
             path="*"
             render={({ history, location }) => {
              testHistory = history;
              testLocation = location;
              return null;
            }} />
            </Routes>
      </MemoryRouter>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});