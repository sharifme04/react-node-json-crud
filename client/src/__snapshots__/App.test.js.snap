// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot data renders 1`] = `
<div
  className="container"
>
  <h2
    className="app-title"
  >
    Todo App
  </h2>
  <hr />
  <div>
    <div
      className="input-group d-flex justify-content-between"
    >
      <a
        href="/todo/addnew"
        onClick={[Function]}
      >
        <button
          className="btn btn-primary"
          type="button"
        >
          Add New
        </button>
      </a>
      <div
        className="search-field"
      >
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="search by title"
          type="text"
          value=""
        />
      </div>
      <div />
    </div>
    <hr />
    <table
      className="table table-striped"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            ID
          </th>
          <th
            scope="col"
          >
            Title
          </th>
          <th
            scope="col"
          >
            Description
          </th>
          <th
            scope="col"
          >
            Date
          </th>
          <th
            scope="col"
          >
            Edit
          </th>
          <th
            scope="col"
          >
            Delete
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`snapshot renders when navigate add route 1`] = `
<div
  className="container"
>
  <h2
    className="app-title"
  >
    Todo App
  </h2>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label>
        Title:
      </label>
      <input
        className="form-control"
        name="title"
        onChange={[Function]}
        type="label"
        value=""
      />
    </div>
    <div
      className="form-group mt-3"
    >
      <label>
        Description:
      </label>
      <input
        className="form-control"
        name="description"
        onChange={[Function]}
        type="label"
        value=""
      />
    </div>
    <div
      className="mt-4"
    />
    <input
      className="btn btn-primary"
      type="submit"
      value="Add"
    />
  </form>
</div>
`;

exports[`snapshot renders when navigate edit route 1`] = `
<div
  className="container"
>
  <h2
    className="app-title"
  >
    Todo App
  </h2>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label>
        Title:
      </label>
      <input
        className="form-control"
        name="title"
        onChange={[Function]}
        type="label"
        value=""
      />
    </div>
    <div
      className="form-group mt-3"
    >
      <label>
        Description:
      </label>
      <input
        className="form-control"
        name="description"
        onChange={[Function]}
        type="label"
        value=""
      />
    </div>
    <div
      className="mt-4"
    />
    <input
      className="btn btn-info"
      type="submit"
      value="Update"
    />
  </form>
</div>
`;
